### DISCUSSION TITLE
Is it planned/doable to make a tower layer/service adapter ?

### DISCUSSION THREAD

**Discussion by Totodore** [rwf2/Rocket#2650] [Ideas] [answered]
Because a lot of libraries are based on tower layer/services, would it be possible to make an adapter that would map rocket middlewares to tower layers and map rocket request/response to http request/response ?

If it is doable, would it be an external library or a feature ?

**Comment by SergioBenitez** [comment, marked as answer]
We can definitely have Rocket accept tower layers/services, and that is indeed planned. It requires implementing early termination first, but this is planned for 0.6.

The other direction is likely better answered elsewhere.
